src/main/java/org/mule/extension/file/internal/lock/FileChannelPathLock.java

```diff
@@ -72,6 +76,48 @@ public boolean tryLock() {
     }
   }
 
+  /**
+   * Due to the limitations of the {@link FileChannel} class, the only way to provided a waiting lock with a specific
+   * timeout that works for 2 or more concurrent locking operations in the same JVM (i.e. in the same Mule App) is to
+   * perform a busy wait loop for the timeout duration. To avoid numerous requests in the loop, on each iteration
+   * the thread is called to sleep for 1/20th of the duration of the timeout.
+   *
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean tryLock(Long timeout) {
+    long nanoTimeout = timeout < 0 ? 0 : TimeUnit.MILLISECONDS.toNanos(timeout);
+    long startTime = System.nanoTime();
+    do {
+      try {
+        lock = channel.tryLock();
+        if (lock != null) {
+          return isLocked();
```

Comment:
What if the lock is invalid here? you would be returning false without waiting for the whole timeout

Reply:
what do you mean by invalid? that the lock is null? it won't return false, it will try again to obtain the lock in the while loop (it only returns false when catching an Exception).
However, now I noticed that, when the lock is null, it tries to obtain the lock again immediately rather than sleeping for 1/20th of the timeout. Fixed that.